apollo-compiler/src/main/kotlin/com/apollostack/compiler/GraphqlCompiler.kt

```diff
@@ -1,6 +1,51 @@
 package com.apollostack.compiler
 
+import com.apollostack.compiler.ir.Field
+import com.apollostack.compiler.ir.QueryIntermediateRepresentation
+import com.squareup.javapoet.*
+import com.squareup.moshi.Moshi
+import java.io.File
+import javax.lang.model.element.Modifier
+
 open class GraphqlCompiler {
+  private val moshi = Moshi.Builder().build()
+
+  fun write(relativePath: String): Status {
+    val irAdapter = moshi.adapter(QueryIntermediateRepresentation::class.java)
+    val ir = irAdapter.fromJson(File(relativePath).readText())
+    // TODO: Handle multiple or no operations
+    val operation = ir.operations.first()
+    val methods = operation.fields.map { it.toMethodSpec() }
+    val customTypes = operation.fields.filter { it.fields.any() }.distinct().map {
+      // TODO: Also need to recurse into inner types
+      TypeSpec.interfaceBuilder(it.type)
+          .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+          .addMethods(it.fields.map { it.toMethodSpec() })
+          .build()
+    }
+    val typeSpec = TypeSpec.interfaceBuilder(operation.operationName)
+        .addMethods(methods)
+        .addTypes(customTypes)
+        .build()
+    JavaFile.builder("test", typeSpec).build()
```

Comment:
@sav007 good point, forgot to keep track of that TODO. That was exactly what I had in mind.
Let me add an item for that